### ISSUE TITLE
Manual workflow modification is required for new features in n8n-nodes-base.httpRequest

### ISSUE THREAD

**Issue by PinkLolicorn** [n8n-io/n8n#3047]
**Describe the bug**
Recently it was made possible to provide body parameters and others in n8n-nodes-base.httpRequest node where the request method is DELETE (https://github.com/n8n-io/n8n/pull/2900). After updating to the `n8n@0.169.0` I was still missing the option in the UI. Further investigation reveled that new nodes have different structure. 

**To Reproduce**
1. Load provided workflow (available below)
2. Inspect both n8n-nodes-base.httpRequest nodes in the UI

**Expected behavior**
After upgrading, `Body Parameters`, `Headers`, and `Query Parameters` are available in the both `HTTP Request` nodes.

**Environment (please complete the following information):**
 - OS: `Ubuntu 20.04.3 LTS`
 - n8n Version `0.169.0`
 - Node.js Version `bundled (docker)`
 - Database system `default`
 - Operation mode `default`

**Additional context**
Removing `jsonParameters` manually seems to fix the issue.

**Workflow file**
```json
{
  "name": "My workflow",
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        400,
        500
      ]
    },
    {
      "parameters": {
        "requestMethod": "DELETE",
        "url": "https://webhook.site/myurl",
        "options": {},
        "bodyParametersUi": {
          "parameter": [
            {
              "name": "hi",
              "value": "aaa"
            }
          ]
        }
      },
      "name": "HTTP Request",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        840,
        420
      ]
    },
    {
      "parameters": {
        "requestMethod": "DELETE",
        "url": "https://webhook.site/myurl",
        "jsonParameters": "=",
        "options": {}
      },
      "name": "HTTP Request1",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        840,
        600
      ]
    }
  ],
  "connections": {},
  "active": false,
  "settings": {},
  "id": 1
}
```

**Comment by Joffcom** [comment]
Hey @PinkLolicorn,

Which version of n8n did you upgrade from?

**Comment by Joffcom** [comment]
Hey @PinkLolicorn,

This issue can actually be reproduced by adding a new HTTP Request then for JSON/RAW Parameters click the cogs and select "Add Expression" then remove the `{{false}}` and hit the exit then to recover from it click the cogs and select "reset value".

I suspect at some point the JSON being used has been manually edited or maybe an export went wrong which is why the previous version would be handy, Failing that though where did the JSON come from that you used and if you put that same JSON into another version of n8n does it work as expected or do you get the same issue?

**Comment by PinkLolicorn** [comment]
> This issue can actually be reproduced by adding a new HTTP Request then for JSON/RAW Parameters click the cogs and select "Add Expression" then remove the `{{false}}` and hit the exit then to recover from it click the cogs and select "reset value".

Just understood what you mean by that. Yes, I remember now that I was trying to modify that not being sure how to provide request body. So it looks like this isn't a migration issue then?

**Comment by PinkLolicorn** [comment]
Ok, this makes sense, tried just now on `0.169.0`. Created a new HTTP Request node and modified Body Parameters, then I cleared manually the expression and closed the popup. The result is an empty expression with visibly different color.

![image](https://user-images.githubusercontent.com/44551064/162014967-056540c7-9b3d-4167-9eac-ee47e821cf8a.png)

```json
{
  "name": "",
  "nodes": [
    {
      "parameters": {
        "requestMethod": "DELETE",
        "url": "https://google.com",
        "jsonParameters": true,
        "options": {},
        "bodyParametersJson": "="
      },
      "name": "HTTP Request",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        440,
        440
      ]
    }
  ],
  "connections": {},
  "active": false,
  "settings": {}
}
```

By closing the node window and downloading it now, I was able to get `"bodyParametersJson": "="` again. Howver, if the `JSON/RAW Parameters` was unchecked beforehand, the new UI elements show up and downloading results with a file without `bodyParametersJson`.

```json
{
  "name": "",
  "nodes": [
    {
      "parameters": {
        "requestMethod": "DELETE",
        "url": "https://google.com",
        "options": {}
      },
      "name": "HTTP Request",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        440,
        440
      ]
    }
  ],
  "connections": {},
  "active": false,
  "settings": {}
}
```

Pasting the node created in the new version, the one with `bodyParametersJson` works. The schema to the one from previous version is a bit different (`jsonParameters is now named `bodyParametersJson`, where in the new version `jsonParameters` is a boolean).

```json
    {
      "parameters": {
        "requestMethod": "DELETE",
        "url": "https://webhook.site/myurl",
        "jsonParameters": "=",
        "options": {}
      },
      "name": "HTTP Request1",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        840,
        600
      ]
    }
```

**Comment by Joffcom** [comment]
That makes sense so what we do is we still allow older versions to work but we keep the schema the same to stop anything from breaking. It can get confusing at times but some people use things like watchtower to automatically apply updates and we wouldn't want to break their installs.

The different input colour is handy and will happen anytime a field is changed to use an expression as a quick visual way to check.

**Comment by PinkLolicorn** [comment]
Ok, following the trail of type mismatch and empty "=" I removed the expression manually, which resulted in it reverting to boolean switch. So it seems to be possible from the UI to modify older nodes, but really:

> It can get confusing at times

---

Old schema:

![image](https://user-images.githubusercontent.com/44551064/162258562-915a7e5f-97f9-4fd2-b7bb-3bcde9222a23.png)

New schema:

![image](https://user-images.githubusercontent.com/44551064/162258476-0f1748bd-c8b2-4bc3-ba33-a2d22456f485.png)

**Comment by Joffcom** [comment]
Hey @PinkLolicorn,

I think I mentioned previously it can be sorted from the UI when I worked out how to manually reproduce it.

What I still don't know is the version that the export came from which could be handy so I can find out which older versions will be impacted on updates and exports.

The nodes from older versions may not have the features that newer versions have and may have a different schema with different field names so if we just ignored any older node it would result in breaking changes for those who do automatic updates so having it as a manual process is sometimes the best option.

If we can get that older version we can try to see why it is exporting that way and if it has any specific meaning from the older version.

**Comment by PinkLolicorn** [comment]
Tried a couple versions back and there was a checkbox in every one of them, then I finally got your message. I'm not sure if this really was the case, but it looks to be user-error created in the process similar to previously described.

> This issue can actually be reproduced by adding a new HTTP Request then for JSON/RAW Parameters click the cogs and select "Add Expression" then remove the `{{false}}` and hit the exit then to recover from it click the cogs and select "reset value".

I think it was most likely the result of adding the expression without noticing the field is, in fact, a checkbox and not the "JSON/Raw Parameters". I blame dynamic typing and no type validation. 🤣
